src/main/js/field/TableDisplay.js

```diff
@@ -126,66 +126,53 @@ class TableDisplay extends Component {
     }
 
     handleClose() {
-        this.props.onConfigClose();
-        this.refs.table.cleanSelected();
         this.setState({
             currentRowSelected: null
         });
+        this.props.onConfigClose();
+        this.refs.table.cleanSelected();
+        this.updateData();
     }
 
     handleSubmit(event) {
         event.preventDefault();
         event.stopPropagation();
-        const { modificationState } = this.state;
-        if (MODIFICATION_STATE.CREATE === modificationState || MODIFICATION_STATE.COPY === modificationState) {
-            this.props.onConfigSave();
-        } else if (MODIFICATION_STATE.EDIT === modificationState) {
-            this.props.onConfigUpdate();
+        const result = this.props.onConfigSave();
+        let validationState;
+        if (result) {
+            validationState = VALIDATION_STATE.SUCCESS;
+        } else {
+            validationState = VALIDATION_STATE.FAILED;
```

Comment:
Would a ternary operator be allowed here or is it not allowed due to async functions?